ui/src/services/prometheus/fetchMetrics.js

```diff
@@ -2,7 +2,15 @@
 import { queryPrometheus } from './api';
 import { PORT_NODE_EXPORTER } from '../../constants';
 
-export function queryNodeFSUsage(instanceIP: string): Promise<Object> {
+export type Prometheus = {
+  status: 'success' | 'error',
+  data: any, // diff between `Expression queries` and `Querying metadatas`
+  error: ?any,
+  errorType: ?any,
+  warnings: ?Array<String>,
+};
```

Comment:
I think a more precise type would be :

```typescript
type RangeMatrixResult = {
    resultType: "matrix",
    result: {
        metric: {
            [labelName: string]: string
        },
        values: [unixTime: number, sampleValue: string][]
    }[] 
}

type InstantVectorResult = {
    resultType: "vector",
    result: {
        metric: {
            [labelName: string]: string
        },
        value: [unixTime: number, sampleValue: string]
    }[]
}

type ScalarResult = {
    resultType: "scalar",
    result: [unixTime: number, scalarValue: string]
}

type StringResult = {
    resultType: "string",
    result: [unixTime: number, stringValue: string]
}

type PrometheusQueryResult = {
    status: "error",
    errorType: string,
    error: string
} | {
    status: "success",
    data: RangeMatrixResult | InstantVectorResult | ScalarResult | StringResult,
    warnings?: string[]
}
```

Reply:
Thank you! 
I am wondering should we name the item of the `result.values[]`, or the following definition is enough?
```
type RangeMatrixResult = {
  resultType: 'matrix',
  result: {
    metric: {
      [labelName: string]: string,
    },
    values: [number, string][],
  }[],
};
```

Reply:
I think it is enough, no need to name the items.